R/paramatch.r

```diff
@@ -1,49 +1,69 @@
+
+#' @export
+paramatch <- function(dataframe, col, top_n_words = 5, suppress.input = FALSE, blacklist = c()) {
+  # Grab the string vector
+  paragraph_col <- dataframe[[col]]
   # Make sure inputs are valid
-    stopifnot(is.character(input))
-    stopifnot(is.numeric(top_n_words))
-    stopifnot(top_n_words >= 1)
-    stopifnot(top_n_words %% 1 == 0)
-    stopifnot(is.logical(suppress.input))
+  stopifnot(is.character(paragraph_col))
+  stopifnot(is.numeric(top_n_words))
+  stopifnot(top_n_words >= 1)
+  stopifnot(top_n_words %% 1 == 0)
+  stopifnot(is.logical(suppress.input))
   # Standardize the input
-    paragraph <- toupper(input)
-    paragraph <- gsub("[[:punct:]]", " ", paragraph)
-    paragraph <- gsub("[[:space:]]+", " ", paragraph)
-  # Split string into words  
-    allwords <- unlist(strsplit(paragraph, " "))
+  paragraph_col <- toupper(paragraph_col)
+  paragraph_col <- gsub("[[:punct:]]", " ", paragraph_col)
+  paragraph_col <- gsub("[[:space:]]+", " ", paragraph_col)
+  # Munge
+  if(!('top_n' %in% names(inputs))) { # Train # TODO add something in case top N length is differnt from inputs
+    # Find the top N words
+    # Split string into words  
+    allwords <- unlist(strsplit(paragraph_col, " "))
     allwords <- allwords[allwords != ""]
-  # Split into unique words
+    # Split into unique words
     words <- unique(allwords)
-  # Full word matching
+    # Full word matching
     frequency <- data.frame(word = words, 
-                            occurances = sapply(words, function(x) length(grep(paste0("\\<",x,"\\>"), allwords))), 
+                            occurances = sapply(words, function(x) length(grep(paste0("\\<", x, "\\>"), allwords))), 
                             stringsAsFactors = FALSE) 
-  # Ensure that N isn't greater than the total number of unique words
+    # Blacklist
+    frequency <- frequency[!frequency$word %in% blacklist, ]
+    # Ensure that N isn't greater than the total number of unique words d
     if(length(allwords) < top_n_words) {
       message("Note: N is too high, defaulting to the number of unique words")
       top_n_words <- length(words)
     }
-  # Take the top N words
-    top_n <- frequency[order(-frequency$occurances),][1:top_n_words,]
-  # Create the output data frame
-    output <- data.frame(input, sapply(top_n$word,
-      function(x) stingr::str_count(paragraph, paste0("\\<",x,"\\>"))))
-    colnames(output) <- c("input", sapply(top_n$word, function(x) paste0("col_", x)))
+    # Take the top N words
+    top_n <- frequency[order(frequency$occurances, decreasing = TRUE),][1:top_n_words, 1]
+    # Write top_n to the Environment
+    inputs$top_n <<- top_n
+  } else {
+    top_n <- inputs$top_n
+  }
 
-  output[if(suppress.input) {-1} else {TRUE}]
+  # Run the count for the top n words
+  output <- data.frame(sapply(top_n, function(x) stringr::str_count(paragraph_col, paste0("\\<", x, "\\>"))))
+  colnames(output) <- c(sapply(top_n, function(x) paste0("col_", x)))
```

Comment:
Don't need the `c` wrapper.` Also, can you replace all instances of `sapply` with `vapply` (last argument `character(1)`)